Document making a release (#47)
populationgenomics/team-docs: new_repository.md

conda build conda/*
```

### Adding GitHub Action

> You can set up GitHub actions to build the package for you, and upload it to the

Reviewer: ```suggestion
You can set up GitHub actions to build and upload the package to the
Anaconda [CPG channel](https://anaconda.org/cpg/), so it becomes available to install
```

 -O .github/workflows/condarse.yaml
```

Now, whenever you run `git push --tags`, a `Condarise` GitHub workflow will be
triggered, build the package versioned after the corresponding git tag, and upload it to
the Anacodna CPG channel. After that, the package will be available with:

```bash
conda install -c cpg -c conda-forge <my-project>
```

## Making a release

In order to release a new version of a tool that is shipped with conda, you would need

> to bump version with `bump2version` and push the created git tag, and also push the "
> Bump version" commit. Because direct pushes to the `main` branch are disabled by

Reviewer 2: ```suggestion
to bump version with `bump2version` and push the created git tag, and also push the
"Bump version" commit. Because direct pushes to the `main` branch are disabled by
```

> to bump version with `bump2version` and push the created git tag, and also push the "
> Bump version" commit. Because direct pushes to the `main` branch are disabled by

Reviewer 2: ```suggestion
to bump the version with `bump2version` and and push both the newly created
"Bump version" commit and git tag. Because direct pushes to the `main` branch
are disabled by
```

default, you would need to create a branch, and a corresponding pull request against
that branch. Here is the full list of steps:

* Create a branch (the name can be arbitrary as this branch will be automatically
  deleted after the pull request gets merged):

```bash

> git checkout -b release

Reviewer 2: Just to make sure the release branch is in sync with `origin/main`:

```suggestion
git checkout -b release
git fetch origin
git reset --hard origin/main
```

```

* Bump a new version (can be `minor` or `major` instead of `patch`):

```bash
bump2version patch
```

* Push the "Bump version" commit:

```bash
git push
```

* Create a pull request against this `release` branch, and request a review.

* After the pull request is merged, push the tag:

```bash
git push --tags
```

Now you can checkout and update `main`:

```bash
git checkout main
git pull --rebase main
```

After the GitHub Actions job is complete, check out that the package appeared

> at [https://anaconda.org/cpg](https://anaconda.org/cpg)

Reviewer 2: ```suggestion
at [https://anaconda.org/cpg](https://anaconda.org/cpg).
```
